src/features/user/Profile/components/RedeemModal.tsx

```diff
@@ -288,69 +303,89 @@ export default function RedeemModal({
                 {errorMessage && (
                   <span className={styles.formErrors}>{errorMessage}</span>
                 )}
-                <button
-                  id={'redeemModalCont'}
-                  onClick={handleSubmit(redeemCode)}
-                  className="primaryButton"
-                  style={{ maxWidth: '200px', marginTop: '24px' }}
-                >
-                  {isUploadingData ? (
-                    <div className={styles.spinner}></div>
-                  ) : (
-                    t('redeem:addToMyTrees')
-                  )}
-                </button>
+                {afterValidateCodeButton ? (
+                  <button
+                    id={'redeemModalCont'}
+                    onClick={closeRedeem}
+                    className="primaryButton"
+                    style={{ maxWidth: '150px', marginTop: '24px' }}
+                  >
+                    {isUploadingData ? (
+                      <div className={styles.spinner}></div>
+                    ) : (
+                      t('redeem:exitToProfile')
+                    )}
+                  </button>
+                ) : (
+                  ''
+                )}
```

Comment:
When you modify existing code, make sure that the naming used makes sense in this context. For example, the `id={'redeemModalCont'}` does not make sense here, as this is not a CONTINUE action any more. It may make sense to change that to `id={'redeemModalClose'}`

Reply:
removed

Reply:
removed